datadog_checks_base/datadog_checks/base/checks/windows/perf_counters/counter.py

```diff
@@ -0,0 +1,516 @@
+        self, name, config, check, connection, object_name, metric_prefix, counter_selector, tags, custom_transformer
+    ):
+        super().__init__(
+            name, config, check, connection, object_name, metric_prefix, counter_selector, tags, custom_transformer
+        )
+
+        self.path = construct_counter_path(
+            machine_name=self.connection.server, object_name=self.object_name, counter_name=self.name
+        )
+        self.counter_handle = None
+
+    def collect(self):
+        try:
+            value = get_counter_value(self.counter_handle)
+        except pywintypes.error as error:
+            self.handle_counter_value_error(error)
+        else:
+            self.transformer(value, tags=self.tags)
+
+    def refresh(self, _):
+        if self.counter_handle is None:
+            self.counter_handle = self.counter_selector(self.connection.query_handle, self.path)
+
+    def clear(self):
+        # https://docs.microsoft.com/en-us/windows/win32/api/pdh/nf-pdh-pdhremovecounter
+        # http://timgolden.me.uk/pywin32-docs/win32pdh__RemoveCounter_meth.html
+        win32pdh.RemoveCounter(self.counter_handle)
+        self.counter_handle = None
+
+
+class MultiCounter(CounterBase):
+    def __init__(
+        self,
+        name,
+        config,
+        check,
+        connection,
+        object_name,
+        metric_prefix,
+        counter_selector,
+        tags,
+        custom_transformer,
+        tag_name,
+    ):
+        super().__init__(
+            name, config, check, connection, object_name, metric_prefix, counter_selector, tags, custom_transformer
+        )
+
+        self.tag_name = tag_name
+
+        self.average = config.get('average', False)
+        if not isinstance(self.average, bool):
+            raise ConfigTypeError(
+                f'Option `average` for counter `{self.name}` of performance '
+                f'object `{self.object_name}` must be a boolean'
+            )
+
+        self.aggregate_transformer = None
+        self.aggregate = config.get('aggregate', False)
+        if not isinstance(self.aggregate, bool) and self.aggregate != 'only':
+            raise ConfigTypeError(
+                f'Option `aggregate` for counter `{self.name}` of performance '
+                f'object `{self.object_name}` must be a boolean or set to `only`'
+            )
+        elif self.aggregate is not False:
+            if self.metric_type not in NATIVE_TRANSFORMERS:
+                raise ConfigTypeError(
+                    f'Option `aggregate` for counter `{self.name}` of performance object `{self.object_name}` is '
+                    f'enabled so `type` must be set to one of the following: {", ".join(NATIVE_TRANSFORMERS)}'
+                )
+            else:
+                metric_name = f'{self.metric_name}.avg' if self.average else f'{self.metric_name}.sum'
+                self.aggregate_transformer = NATIVE_TRANSFORMERS[self.metric_type](check, metric_name, config)
```

Comment:
Is this where the metric submissions are made? Should we add some debug logging to log out the collected value for each metric name? That could be helpful for troubleshooting.

Reply:
Submission occurs during `.collect()`. Every value basically gets submitted straight to a transformer (which cannot fail). If there is a Windows API error then that will be logged in `.handle_counter_value_error()`.